groups.cabal

```diff
@@ -14,6 +14,6 @@ cabal-version:       >=1.8
 
 library
   exposed-modules:     Data.Group
-  -- other-modules:       
-  build-depends:       base <5
+  -- other-modules:
+  build-depends:       base >= 4.12.0.0 && < 5
```

Comment:
The lower bound is when the `Monoid` instances for `(:*:)` and `(:.:)` appeared.

Reply:
Hmm, I think I'd rather we used CPP so we could still support at least as far back as 7.6 (the version of GHC in debian oldoldstable and CentOS 7). I'm going to finally add a test suite soon so we can check this automatically!

Reply:
New base bound supports GHC 7.6.1 and above, but creates a lot of CPP noise. `{-# LANGUAGE TypeOperators #-}` has been around since GHC 6.8.1 so I didn't try to CPP that away.